src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -41,6 +44,44 @@ def enable_telemetry
       other_client.logout rescue nil
     end
 
+    def enable_extension
+      http_client = VSphereCloud::CpiHttpClient.new
```

Comment:
do you need a separate client for this?

Reply:
Just for the fact that it is a separate operation like adding telemetry configuration and keeping it asynchronous helps to speed up things. It takes sometime to register and find an extension.